Dismiss variants. Fix #605

dnil opened the pull request.

moonso commented:
Wow fast work @dnil ! I will need some time to look this through...

moonso commented on scout/adapter/mongo/event.py:
```diff
@@ -871,6 +871,57 @@ def update_manual_rank(self, institute, case, user, link, variant,
 
+    def update_dismiss_variant(self, institute, case, user, link, variant,
+                           dismiss_variant):
+        """Create an event for updating the manual rank of a variant
```
Update the docstring?

dnil replied:
Ahem, yes. 😊

moonso replied:
😸

moonso commented:
Also @dnil , could you add a test or two? The event tests are pretty straightforward, it would be nice to have for this one as well.

dnil commented:
Added event test. Also got some feedback on the color scheme. Since the info-color is used for e.g. causative transcripts, it is not ideal for marking dismissed variants. Added a new gray to fit the role.

northwestwitch commented:
Nice!

moonso commented:
Hi @dnil, looks nice! One question: It looks like the same event will be created regardless if a variant is dismissed or if it is "reset dismiss". It could be a bit confusing when looking at the history of events for a variant or case. Could you update the verbs perhaps?

dnil commented:
Right; sending an empty list will unset. I thought it simple and robust, but didn't really consider logging.  It has the same granularity as set/unset manual_rank. Has that been a problem?

moonso commented:
Maybe not a problem. Lets go for this! We can change that one later
